src/main/java/org/wildfly/security/audit/AuditEndpoint.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.audit;
+
+import java.io.IOException;
+
+import org.wildfly.common.function.ExceptionBiConsumer;
+
+/**
+ * An endpoint that recieves audit messages.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ */
+public interface AuditEndpoint extends ExceptionBiConsumer<EventPriority, String, IOException> {
```

Comment:
If you're going functional then you should make close() default.  Also extend AutoCloseable any time you have a close() method, or Closeable any time you have one that throws IOException.

If you're not going functional then extending a functional interface is a little weird.